src/Search/PromotionList.js

```diff
@@ -0,0 +1,54 @@
+import React, {Component} from 'react'
+import Promotion from './Promotion.js'
+import PropTypes from "prop-types"
+
+class PromotionList extends Component{
+    constructor(props){
+        super(props);
+        this.state = {
+          query: '',
+    
+        }
+      }
+    
+    handleInput = e => {
+        this.setState({query: e.target.value})
+    }
+
+    render(){
+        const filteredPromotions = this.props.promotions.filter(
+            (promotion) => {
+                return promotion.name.toLowerCase().indexOf(this.state.query.toLowerCase()) !== -1;
+            }
+        );
```

Comment:
this can be rewritten as following:
```
        const { query } = this.state;
        const { promotions } = this.props;
        
        const filteredPromotions = promotions.filter(
            ({ name })  => {
                return name.toLowerCase().indexOf(query.toLowerCase()) > -1;
            }
        );
```